plugins/snippets/luasnip/default.nix

```diff
@@ -8,6 +8,51 @@
 with lib;
 let
   cfg = config.plugins.luasnip;
+
+  loaderSubmodule = types.submodule {
+    options = {
+      lazyLoad = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether or not to lazy load the snippets
+        '';
+      };
+
+      # TODO: add option to also include the default runtimepath
+      paths = mkOption {
+        default = null;
+        type =
+          with types;
+          nullOr (oneOf [
+            str
+            path
+            helpers.nixvimTypes.rawLua
+            (listOf (oneOf [
+              str
+              path
+              helpers.nixvimTypes.rawLua
+            ]))
+          ]);
+      };
+
+      exclude = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        description = ''
+          List of languages to exclude, by default is empty.
+        '';
+      };
+
+      include = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        description = ''
+          List of languages to include, by default is not set.
+        '';
+      };
```

Comment:
```suggestion
      include = helpers.mkNullOrOption (with helpers.nixvimTypes; maybeRaw (listOf str)) ''
        List of languages to include, by default is not set.
      '';
```

Reply:
Should it be `maybeRaw (listOf (maybeRaw str))`?

Reply:
Haha I knew you would say that !
(it's fine for it to be only a `listOf str`, but yes, it could also accept raw element).